Test failure: TestLtrMsmarcoDocument and TestLtrMsmarcoPassage

```
======================================================================
FAIL: test_reranking (test_ltr_msmarco_document.TestLtrMsmarcoDocument)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch2/jimmylin/pyserini/integrations/sparse/test_ltr_msmarco_document.py", line 59, in test_reranking
    self.assertAlmostEqual(float(mrr),0.3090492928920076, delta=0.000001)
AssertionError: 0.30906234846241004 != 0.3090492928920076 within 1e-06 delta (1.3055570402431549e-05 difference)

======================================================================
FAIL: test_reranking (test_ltr_msmarco_passage.TestLtrMsmarcoPassage)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch2/jimmylin/pyserini/integrations/sparse/test_ltr_msmarco_passage.py", line 53, in test_reranking
    self.assertAlmostEqual(float(mrr),0.24709612498294367, delta=0.000001)
AssertionError: 0.24723580979669724 != 0.24709612498294367 within 1e-06 delta (0.00013968481375356556 difference)

```

Seems like the underlying dependencies have changed a bit... this seems like an issue of brittle tests; we probably don't need to hit the score within a delta of 1e-06.

@yuki617 @stephaniewhoo Okay if we dial down the delta a bit? I'll take care of sending PR.

> ```
> ======================================================================
> FAIL: test_reranking (test_ltr_msmarco_document.TestLtrMsmarcoDocument)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/scratch2/jimmylin/pyserini/integrations/sparse/test_ltr_msmarco_document.py", line 59, in test_reranking
>     self.assertAlmostEqual(float(mrr),0.3090492928920076, delta=0.000001)
> AssertionError: 0.30906234846241004 != 0.3090492928920076 within 1e-06 delta (1.3055570402431549e-05 difference)
> 
> ======================================================================
> FAIL: test_reranking (test_ltr_msmarco_passage.TestLtrMsmarcoPassage)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/scratch2/jimmylin/pyserini/integrations/sparse/test_ltr_msmarco_passage.py", line 53, in test_reranking
>     self.assertAlmostEqual(float(mrr),0.24709612498294367, delta=0.000001)
> AssertionError: 0.24723580979669724 != 0.24709612498294367 within 1e-06 delta (0.00013968481375356556 difference)
> ```
> 
> 
>      
>   
> 
> Seems like the underlying dependencies have changed a bit... this seems like an issue of brittle tests; we probably don't need to hit the score within a delta of 1e-06.
> 
> 
> @yuki617 @stephaniewhoo Okay if we dial down the delta a bit? I'll take care of sending PR.

Yes, lgtm! Thx

> ```
> ======================================================================
> FAIL: test_reranking (test_ltr_msmarco_document.TestLtrMsmarcoDocument)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/scratch2/jimmylin/pyserini/integrations/sparse/test_ltr_msmarco_document.py", line 59, in test_reranking
>     self.assertAlmostEqual(float(mrr),0.3090492928920076, delta=0.000001)
> AssertionError: 0.30906234846241004 != 0.3090492928920076 within 1e-06 delta (1.3055570402431549e-05 difference)
> 
> ======================================================================
> FAIL: test_reranking (test_ltr_msmarco_passage.TestLtrMsmarcoPassage)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/scratch2/jimmylin/pyserini/integrations/sparse/test_ltr_msmarco_passage.py", line 53, in test_reranking
>     self.assertAlmostEqual(float(mrr),0.24709612498294367, delta=0.000001)
> AssertionError: 0.24723580979669724 != 0.24709612498294367 within 1e-06 delta (0.00013968481375356556 difference)
> ```
> 
> Seems like the underlying dependencies have changed a bit... this seems like an issue of brittle tests; we probably don't need to hit the score within a delta of 1e-06.
> 
> @yuki617 @stephaniewhoo Okay if we dial down the delta a bit? I'll take care of sending PR.

Yes, the difference is fairly small. Let's dial down the delta. Thanks!

--- a/integrations/sparse/test_ltr_msmarco_document.py
+++ b/integrations/sparse/test_ltr_msmarco_document.py
@@ -56,7 +56,8 @@ class TestLtrMsmarcoDocument(unittest.TestCase):
         result = subprocess.check_output(f'python tools/scripts/msmarco/msmarco_doc_eval.py --judgments tools/topics-and-qrels/qrels.msmarco-doc.dev.txt --run ltr_test/{outp_tsv}', shell=True).decode(sys.stdout.encoding)
         a,b = result.find('#####################\nMRR @100:'), result.find('\nQueriesRanked: 5193\n#####################\n')
         mrr = result[a+32:b]
-        self.assertAlmostEqual(float(mrr),0.3090492928920076, delta=0.000001)
+        # See https://github.com/castorini/pyserini/issues/951
+        self.assertAlmostEqual(float(mrr), 0.3091, delta=0.0001)
         rmtree('ltr_test')
 
 if __name__ == '__main__':

--- a/integrations/sparse/test_ltr_msmarco_passage.py
+++ b/integrations/sparse/test_ltr_msmarco_passage.py
@@ -50,7 +50,8 @@ class TestLtrMsmarcoPassage(unittest.TestCase):
         result = subprocess.check_output(f'python tools/scripts/msmarco/msmarco_passage_eval.py tools/topics-and-qrels/qrels.msmarco-passage.dev-subset.txt ltr_test/{outp}', shell=True).decode(sys.stdout.encoding)
         a,b = result.find('#####################\nMRR @10:'), result.find('\nQueriesRanked: 6980\n#####################\n')
         mrr = result[a+31:b]
-        self.assertAlmostEqual(float(mrr),0.24709612498294367, delta=0.000001)
+        # See https://github.com/castorini/pyserini/issues/951
+        self.assertAlmostEqual(float(mrr), 0.2472, delta=0.0001)
         rmtree('ltr_test')
 
 if __name__ == '__main__':

